File: Fikra.Model/Entities/Task.cs
using Fikra.Model.Entities.Enums;
using Fikra.Model.Interfaces;
using System;
using System.Collections.Generic;
using System.Text;

namespace Fikra.Model.Entities
{
    public class Task : IEntity<Guid>
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }
        public bool IsDeleted { get; set; }
		public DateTime? Due { get; set; }
        public Status Status { get; set; }
        public Priority Priority { get; set; }
        public Effort Effort { get; set; }
        public ICollection<TaskComment> Comments { get; set; }
    }
}

File: Fikra.Ng/wwwroot/fikra/src/app/models/task.ts
import { Effort } from './effort';
import { Priority } from './priority';
import { Status } from './status';
import { Comment } from './comment';

export class Task {
    Id: string;
    CreatedOn: Date;
    ModifiedOn: Date;
    Due: Date;
    Name: string;
    Status: Status;
    Priority: Priority;
    Effort: Effort;
    Comments: Array<Comment>;

    constructor() {
        this.CreatedOn = new Date();
        this.ModifiedOn = new Date();
        this.Due = null;
        this.Name = 'New Task';
        this.Status = Status.New;
        this.Priority = Priority.Low;
        this.Effort = null;
        this.Comments = [];
    }
}